src/libcore/ops.rs

```diff
@@ -1539,6 +1539,10 @@ impl<Idx: PartialOrd<Idx>> Range<Idx> {
 ///
 /// See the [`contains()`](#method.contains) method for its characterization.
 ///
+/// Note: No overflow checking is done for the iterator implementation; if you
+/// use an integer range and the integer overflows, it will panic in debug mode
+/// and create an endless loop in release mode.
```

Comment:
No, it's always an endless loop (if it doesn't panic), the implementation does not have a path that returns `None`: https://github.com/tbu-/rust/blob/ad9f9a563ed82df5ea82bc32497dd986614bdee7/src/libcore/iter.rs#L4642-L4654.